File: src/7.整数反转.js
/*
 * @lc app=leetcode.cn id=7 lang=javascript
 *
 * [7] 整数反转
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var sum=0;
    while(x!=0){
        sum=sum*10+x%10;
        x=(x/10)|0;
    }
    return (sum | 0) === sum ? sum : 0;
};
// @lc code=end

File: src/7.整数反转.go
/*
 * @lc app=leetcode.cn id=7 lang=golang
 *
 * [7] 整数反转
 */
package leetcode

import "math"

// @lc code=start
func reverse(x int) int {
	ans := 0

	for x!=0 {
		if math.MinInt32/10 > ans || ans > math.MaxInt32/10 {
			return 0
		}
		digit := x%10
		x/=10
		ans = ans*10 + digit
	}

	return ans
}
// @lc code=end

